rsconf/component/elasticsearch.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+u"""create elasticsearch configuration
+:copyright: NA
+:license: NA
+"""
+import os
+from __future__ import absolute_import, division, print_function
+from pykern.pkcollections import PKDict
+from rsconf import component
+from pykern import pkio
+from pathlib import Path
+from jinja2 import FileSystemLoader, Environment
+
+# helper function for rendering elasticsearch config
+def render_config(host_name, host_ip, host_port):
+    template_dir = "/home/vagrant/src/radiasoft/rsconf/rsconf/package_data/elasticsearch"
+    env = Environment(loader=FileSystemLoader(template_dir))
+    template = env.get_template("elasticsearch.yml.jinja")
+    return template.render(host_name=host_name, host_ip=host_ip, host_port=host_port)
+    
+# helper function for creating elasticsearch API command for creating a user
+def create_user(username, password, roles, full_name, email):
+    template_dir = "/home/vagrant/src/radiasoft/rsconf/rsconf/package_data/elasticsearch"
+    env = Environment(loader=FileSystemLoader(template_dir))
+    template = env.get_template("user_request.txt.jinja")
+    return template.render(username=username, password=password, roles=roles, full_name=full_name, email=email)
+
+class T(component.T):
+
+    def internal_build(self): # Difference between internal_build and internal_build_compile ??
+        self.buildt.require_component('base_all') # componants generally need base_all
+
+        # print to test component is being touched
+        self.append_root_bash('echo "Elasticsearch touched"')
+        
+        # install/update java jdk
+        self.append_root_bash('rsconf_yum_install java-1.8.0-openjdk')
+        
+        # download + install public signing key
+        self.append_root_bash('sudo rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch')
```

Comment:
rsconf runs as root so no sudo. Importing a key is something you only do once as well as adding the repo. We should add something to rsconf.sh like `rsconf_yum_import_key`. The repo file itself would be installed as a normal file (install_resource would work).